### ISSUE TITLE
Support triggers and rules under package?

### ISSUE THREAD

**Issue by jthomas** [openwhisk/openwhisk#1825]
Packages are a convenient mechanism to group related openwhisk resources into "services".

According to the documentation, packages can only currently contain actions and feeds. 

Would it be possible to extend this to include triggers and rules?

**Comment by rabbah** [comment]
This is easier to do/planned as part of this https://github.com/openwhisk/openwhisk/issues/219.

**Comment by jthomas** [comment]
Is this still planned?

**Comment by rabbah** [comment]
Yes we should do this. Hasn’t been a priority though with a lot of work on performance and scale out.

**Comment by mrutkows** [comment]
@rabbah @jthomas agree. We have been experiencing name collisions with wskdeploy tests as devs copy names from one test file to another and the results can be quite frustrating as we automate.

**Comment by jthomas** [comment]
Good point @mrutkows. I have the same issue in The Serverless Framework. 
https://github.com/serverless/serverless-openwhisk/issues/29

**Comment by rabbah** [comment]
As noted here: https://github.com/apache/incubator-openwhisk/issues/3247#issuecomment-363112438 there are some nuances we need to address, but otherwise allowing a trigger or rule to be nested in a package is plausible.

**Comment by ScottChapman** [comment]
Yea, I think it would be wise to bump the priority of this. First it just feel arbitrary. And second and you start getting more and more integrations it will become more and more problematic for users.

**Comment by yumeforever** [comment]
Any news on this thread? Any other suggested threads to follow?

**Comment by rabbah** [comment]
The other relevant bit is https://github.com/apache/incubator-openwhisk/issues/219#issuecomment-272951007 but there isn't any progress yet as far as I know on this.

**Comment by akrabat** [comment]
Discovering that triggers and rules couldn't be in packages threw me completely! This would be really handy.

**Comment by jthomas** [comment]
Still hoping for this in 2019 🙏

**Comment by csantanapr** [comment]
no one is looking into this, it's open for grabs
